developer-extension/src/panel/components/tabs/eventsTab/facetList.tsx

```diff
@@ -0,0 +1,170 @@
+import { Box, Card, Checkbox, Collapse, Flex, Text } from '@mantine/core'
+import React from 'react'
+import type { ExcludedFacetValues, FacetRegistry } from '../../../hooks/useEvents'
+import type { Facet } from '../../../facets.constants'
+import { FACETS, FacetValue } from '../../../facets.constants'
+import { BORDER_RADIUS, tabsListBorder } from '../../../uiUtils'
+
+export function FacetList({
+  facetRegistry,
+  excludedFacetValues,
+  onExcludedFacetValuesChange,
+}: {
+  facetRegistry: FacetRegistry
+  excludedFacetValues: ExcludedFacetValues
+  onExcludedFacetValuesChange: (newExcludedFacetValues: ExcludedFacetValues) => void
+}) {
+  return (
+    <FacetField
+      facet={FACETS[0]}
+      depth={0}
+      facetRegistry={facetRegistry}
+      excludedFacetValues={excludedFacetValues}
+      onExcludedFacetValuesChange={onExcludedFacetValuesChange}
+    />
+  )
+}
+
+function FacetField({
+  facet,
+  depth,
+  facetRegistry,
+  excludedFacetValues,
+  onExcludedFacetValuesChange,
+}: {
+  facet: Facet
+  depth: number
+  facetRegistry: FacetRegistry
+  excludedFacetValues: ExcludedFacetValues
+  onExcludedFacetValuesChange: (newExcludedFacetValues: ExcludedFacetValues) => void
+}) {
+  const facetValueCounts = facetRegistry.getFacetValueCounts(facet.id)
+
+  return (
+    <Box>
+      {facet.label && (
+        <Text pt="xs" color="dimmed" weight="bold">
+          {facet.label}
+        </Text>
+      )}
+
+      {Array.from(facetValueCounts.entries(), ([facetValue, facetValueCount]) => (
+        <FacetValue
+          key={facetValue}
+          facet={facet}
+          facetValue={facetValue}
+          facetValueCount={facetValueCount}
+          depth={depth}
+          facetRegistry={facetRegistry}
+          excludedFacetValues={excludedFacetValues}
+          onExcludedFacetValuesChange={onExcludedFacetValuesChange}
+        />
+      ))}
+    </Box>
+  )
+}
+
+const SPACE_BETWEEN_CHECKBOX = 4 // the smallest size defined by mantine ("xs") is 10px, which is a bit too much
+
+function FacetValue({
+  facet,
+  facetValue,
+  facetValueCount,
+  depth,
+  facetRegistry,
+  excludedFacetValues,
+  onExcludedFacetValuesChange,
+}: {
+  facet: Facet
+  facetValue: FacetValue
+  facetValueCount: number
+  depth: number
+  facetRegistry: FacetRegistry
+  excludedFacetValues: ExcludedFacetValues
+  onExcludedFacetValuesChange: (newExcludedFacetValues: ExcludedFacetValues) => void
+}) {
+  const isTopLevel = depth === 0
+  const isSelected = !excludedFacetValues[facet.id] || !excludedFacetValues[facet.id].includes(facetValue)
+  const value = (
+    <Flex justify="space-between" mt={isTopLevel ? 'xs' : SPACE_BETWEEN_CHECKBOX}>
+      <Checkbox
+        label={facet.values?.[facetValue]?.label ?? facetValue}
+        checked={isSelected}
+        onChange={() => {
+          onExcludedFacetValuesChange(toggleExcludedFacetValue(facet, excludedFacetValues, facetValue))
+        }}
+      />
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Would it be complicated to have the same behaviour than the datadog facet list with "toggle" or "only" depending on if we interact with the checkbox or the label?

Reply:
It depends. The particularity here is that we have dynamic facet values, so we don't know the values in advance (this is why I found it easier to store a list of facet values to exclude, instead of having a list of facet value to keep).

We could have a "only" button that exclude all facet existing values, but then some new values can come and it wouldn't be an "only" anymore. This could be fairly easy to implement, and bring enough value for our needs.

An better alternative could be to have a more involved filter strategy that selects only a given value even when other values are coming. It would be a bit more complex to implement.

In any case, it will be part of a future PR :)